src/config.rs

```diff
@@ -3,11 +3,35 @@
 
 use yaml_rust::Yaml;
 use yaml_rust::YamlLoader;
+use yaml_rust::ScanError;
 
 use std::env::home_dir;
 use std::fs::File;
+use std::io;
 use std::io::Read;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
+use std::error;
+use std::fmt;
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    ///List of servers
+    pub servers: Vec<Server>,
+    ///Defaults: see definition for Defaults struct
+    pub defaults: Defaults,
+    ///Path to store chatlogs
+    pub logs: PathBuf,
+}
+
+impl Config {
+    pub fn new<P: AsRef<Path>>(servers: Vec<Server>, defaults: Defaults, logs: P) -> Config {
```

Comment:
Why not use the owned type as type of `logs` here if you're going to convert it to the owned type anyway? That way you avoid allocations in some cases because if the call site can move it to this function it wouldn't have to pass a reference only to copy it here.